tests/support/mockRegistryProvider.ts

```diff
@@ -0,0 +1,58 @@
+import createRenderMixin, { RenderMixin, RenderMixinState } from '../../src/mixins/createRenderMixin';
+import Promise from 'dojo-shim/Promise';
+import { Child } from '../../src/mixins/interfaces';
+import { ComposeFactory } from 'dojo-compose/compose';
+import Map from 'dojo-shim/Map';
+import WeakMap from 'dojo-shim/WeakMap';
+
+export const widgetMap: Map<string | symbol, Child> = new Map<string | symbol, Child>();
+const widgetIdMap: WeakMap<Child, string | symbol> = new WeakMap<Child, string | symbol>();
+
+const widgetIds: string[] = [ 'widget1', 'widget2', 'widget3', 'widget4' ];
+let widgetUID = 5;
+
+function reset() {
+	widgetMap.clear();
+	widgetUID = 5;
+	widgetRegistry.stack = [];
+
+	widgetIds.forEach((id) => {
+		const widget = createRenderMixin({ id });
+		widget.own({
+			destroy() {
+				widgetMap.delete(id);
+				widgetIdMap.delete(widget);
+			}
+		});
+		widgetMap.set(id, widget);
+		widgetIdMap.set(widget, id);
+	});
+}
+
+const widgetRegistry = {
+	reset,
+	stack: <(string | symbol)[]> [],
+	get(id: string | symbol): Promise<RenderMixin<RenderMixinState>> {
+		widgetRegistry.stack.push(id);
+		return Promise.resolve(widgetMap.get(id));
```

Comment:
reject, right?  But I guess why do we reject with an error?  Is there anyone who really *needs* that?